src/e3/anod/qualifiers_manager.py

```diff
@@ -219,6 +227,122 @@ def repr(self, value: str | bool, hash_pool: list[str] | None) -> str:
             return ""
 
 
+class KeySetDeclaration(QualifierDeclaration):
+    def __init__(
+        self,
+        origin: str,
+        name: str,
+        description: str,
+        repr_in_hash: bool = False,
+        repr_name: str | None = None,
+        repr_omit_key: bool = False,
+        default: str | None = None,
+        choices: list[str] | None = None,
+    ) -> None:
+        """Initialize a key-set qualifier declaration.
+
+        :param origin: a string giving the origin of the declaration
+            (used in error messages)
+        :param name: see QualifierDeclaration.__init__
+        :param description: see QualifierDeclaration.__init__
+        :param repr_in_hash: see QualifierDeclaration.__init__
+        :param repr_name: see QualifierDeclaration.__init__
+        :param repr_omit_key: if True discard qualifier name in string representation
+        :param default: default value for the qualifier
+        :param choices: list of valid value for the qualifier
+        """
+        super().__init__(
+            origin=origin,
+            name=name,
+            description=description,
+            repr_in_hash=repr_in_hash,
+            repr_name=repr_name,
+        )
+        self.repr_omit_key = repr_omit_key
+
+        # Check if the default is valid
+        default_values: frozenset[str] | None = None
+        if default is not None:
+            default_values = frozenset(default.split(";"))
```

Comment:
Define a constant for the ; separator, it should make things more understandable